src/Microsoft.AspNet.Razor/Generator/TagHelperCodeGenerator.cs

```diff
@@ -66,6 +66,7 @@ public override void GenerateStartBlockCode(Block target, CodeGeneratorContext c
 
                 attributes[attribute.Key] = new ChunkBlock
                 {
+                    Association = first?.Association,
```

Comment:
because I didn't notice `target` was a `Span`.  I'll see if that changes anything....